frontend/src/js/components/PageViewer/VirtualScroll.tsx

```diff
@@ -180,23 +215,25 @@ export const VirtualScroll: FC<VirtualScrollProps> = ({
     <div ref={viewport} className={styles.scrollContainer} onScroll={throttledSetPageRangeFromScrollPosition}>
       <div className={styles.pages} style={{ height: totalPages * pageHeight }}>
         {renderedPages.map((page) => (
-          <div
-            key={page.pageNumber}
-            style={{
-              top: (page.pageNumber - 1) * pageHeight,
-              transform: `rotate(${rotation}deg)`,
-            }}
-            className={styles.pageContainer}
-          >
-            <Page
-              focusedFindHighlightId={focusedFindHighlight?.id}
-              focusedSearchHighlightId={focusedSearchHighlight?.id}
-              pageNumber={page.pageNumber}
-              getPagePreview={page.getPagePreview}
-              getPageData={page.getPageData}
-            />
-          </div>
-        ))}
+            <div
+              key={page.pageNumber}
+              style={{
+                top: (page.pageNumber - 1) * pageHeight,
+                transform: `rotate(${rotation}deg)`,
+                left: `${scale > 1 ? '0' : ''}`,  
+              }}
```

Comment:
I had to set the left to 0 in here, because when the pages are landscape the page left side could go out of the container box
![ezgif-5-80722b1ef1](https://github.com/guardian/giant/assets/15894063/a866c4e8-86e6-4e93-b004-a73b8d71acf7)

Reply:
I'm not totally sure this is the right approach. Would need to pair on it but I feel like the right solution is keep the page centred, but be able to scroll leftwards if it's gone off the edge. There are lots of intermediate wrapping divs though so not quite sure what the right styling is to achieve that!

Reply:
Thanks Jo, I created a function `setLeftScrollToMiddle` that would set the scroll to middle every-time the zooming is changed. I had to use a 1 second delay, because we need to calculate the scroll position after the new canvases have been created and we know the final scrollWidth. What do you think?

https://github.com/guardian/giant/pull/112/files#diff-5080e96f143544d1f02e141bb59b417dc6f40b6cc8c9693bdd75688c2c913b97R103

Reply:
This is a tricky one! I've had a bit more of a think and play around with this. Separating out the two issues:

### Overflow
`left: 0` isn't ideal because it means the page is no longer centred when at normal zoom, which is a regression from existing functionality.

Something was bothering me about the fact that you had to do this, because there should definitely be a way to make the `.scrollContainer` provide a horizontal scrollbar the respects the width of its children, e.g. https://jsfiddle.net/14fdgea0/

After playing around, I realised it's the flex layouts combined with absolute positioning that is [adding confusion](https://css-tricks.com/flexbox-and-absolute-positioning/). I've found that removing `display: flex` on the `.pages` element makes it behave correctly with respect to overflow.

The thing is, we're using flex to horizontally centre the pages so with that solution, we then end up back at the pages no longer being centred.

So the right solution will be to remove the absolute positioning and do everything with flex, or to remove flex and do the horizontal centre with a different technique (e.g. translate: https://medium.com/front-end-weekly/absolute-centering-in-css-ea3a9d0ad72e). I think the latter may be easier!

### Scroll
Using the 1 second delay is a nice way to prove out the logic (without getting bogged down in how to make it happen after render), but I don't think this would be a great user experience to move the page around under them. 

And thinking about it a bit more, centring the scrollbar is not actually desired behaviour while zooming: if you're trying to zoom to look at something in more detail, you want it to remain in view, and not lose your bearings by having the scrollbar yanked away to somewhere else.

Fortunately, I think the right solution to the overflow issue might make the scrollbar issue irrelevant: the browser's default behaviour should give us what we want, once we are able to get rid of the `left: 0`

Reply:
**also...** I've realised that this issue isn't actually one that you created with the zoom functionality. It already existed due to the weird combo of flexbox and absolute positioning, but the zoom made it more obvious.

For instance, on PROD right now if you shrink the window, the left of the doc is cut off, even though the scrollbar is at the far left!

@itsibitzi do you have any memories of why you used absolute positioning with flex?

<img width="674" alt="Screenshot 2023-07-25 at 12 42 38" src="https://github.com/guardian/giant/assets/5122968/1c184ee2-976c-47f0-8b3b-02934df2bea6">